index.js

```diff
@@ -269,10 +269,26 @@ var PgDriver = Base.extend({
         var constraint = [],
             cb;
 
+        if ((spec.unique || spec.primaryKey) && spec.autoIncrement) {
+          switch (spec.dataType) {
+                case type.BIGINT:
+                    constraint.push('BIGSERIAL');                    
+                    break;
+                case type.SMALLINT:
+                    constraint.push('SMALLSERIAL');
+                    break;
+                default:
+                    constraint.push('SERIAL');
+                    break;
+            }          
+        }
+        else {
+          if (spec.autoIncrement) {
+            throw 'Auto increment column must be unique or primary key';
```

Comment:
Yeah, but the database let db-migrate handle the failure :)

And `createColumnDef` gets overwritten here that is true, so there could be the case that a base function could call this one and wouldn't expect a promise. But that would be easy to fix though. So yes it would probably need also some changes on the base driver. But rather small ones, as we wouldn't make `createColumnDef` a promise in the base driver, as this would possibly break all other drivers, but simply change all methods using this function, to know that this could return a promise by wrapping the result in a `Promise.resolve()`